src/main/java/org/jenkinsci/plugins/workflow/support/actions/LogActionImpl.java

```diff
@@ -110,8 +101,8 @@
     private String charset;
 
     private LogActionImpl(FlowNode parent, Charset charset) {
-        if (!isRunning(parent)) {
-            throw new IllegalStateException("cannot start writing logs to a finished node " + parent);
+        if (!parent.isActive()) {
+            LOGGER.log(Level.WARNING, null, new IllegalStateException("cannot start writing logs to a finished node " + parent + " " + parent.getDisplayFunctionName() + " in " + parent.getExecution()));
```

Comment:
Mixed feelings on this @jglick -- think it's better to throw the exception here because then we can't create an invalid LogActionImpl.  Plus AFAICT the exception wouldn't be otherwise swallowed, and we can always try/catch it at the single constructor invocation if needed.